src/test/javascript/spec/component/text-assessment-dashboard/text-assessment-dashboard.component.spec.ts

```diff
@@ -0,0 +1,257 @@
+        expect(component.filteredSubmissions).toEqual([textSubmission]);
+    }));
+
+    it('should not get Submissions', () => {
+        const getTextSubmissionStub = jest.spyOn(textSubmissionService, 'getTextSubmissionsForExerciseByCorrectionRound');
+        getTextSubmissionStub.mockReturnValue(of(new HttpResponse({ body: [], headers: new HttpHeaders() })));
+        const isAtLeastInstructorInCourseStub = jest.spyOn(accountService, 'isAtLeastInstructorInCourse');
+        isAtLeastInstructorInCourseStub.mockReturnValue(true);
+        const findExerciseStub = jest.spyOn(exerciseService, 'find');
+        findExerciseStub.mockReturnValue(of(new HttpResponse({ body: textExercise, headers: new HttpHeaders() })));
+        component.exercise = textExercise;
+        // call
+        component.ngOnInit();
+
+        // check
+        expect(findExerciseStub).toHaveBeenCalledTimes(1);
+        expect(getTextSubmissionStub).toHaveBeenCalledWith(textExercise.id, { submittedOnly: true });
+        expect(component.submissions).toEqual([]);
+        expect(component.filteredSubmissions).toEqual([]);
+    });
+
+    it('should update filtered submissions', () => {
+        // setup
+        component.ngOnInit();
+        component.updateFilteredSubmissions([textSubmission]);
+        // check
+        expect(component.filteredSubmissions).toEqual([textSubmission]);
+    });
+
+    it('should cancel assessment', fakeAsync(() => {
+        // test cancelAssessment
+        const windowSpy = jest.spyOn(window, 'confirm').mockReturnValue(true);
+        const cancelAssessmentStub = jest.spyOn(textAssessmentService, 'cancelAssessment').mockReturnValue(of());
+        component.exercise = textExercise;
+
+        // call
+        component.cancelAssessment(textSubmission);
+        tick();
+
+        // check
+        expect(cancelAssessmentStub).toHaveBeenCalledTimes(1);
+        expect(cancelAssessmentStub).toHaveBeenCalledWith(textSubmission.participation.id, textSubmission.id);
+        expect(windowSpy).toHaveBeenCalledTimes(1);
+    }));
+
+    it('should sort rows', () => {
+        // test cancelAssessment
+        const sortServiceSpy = jest.spyOn(sortService, 'sortByProperty');
+        component.predicate = 'predicate';
+        component.reverse = false;
+        component.submissions = [textSubmission];
+        component.sortRows();
+
+        expect(sortServiceSpy).toHaveBeenCalledTimes(1);
+        expect(sortServiceSpy).toHaveBeenCalledWith([textSubmission], 'predicate', false);
+    });
+
+    it('should get the assessment type of a result', () => {
+        const result = { id: 55, assessmentType: AssessmentType.SEMI_AUTOMATIC };
+        expect(component.assessmentTypeTranslationKey(result)).toBe(`artemisApp.AssessmentType.${result.assessmentType}`);
+        expect(component.assessmentTypeTranslationKey(undefined)).toBe(`artemisApp.AssessmentType.null`);
+    });
+
+    describe('should get assessment link', () => {
```

Comment:
I would remove the ``describe``-wrapper here as it doesn't serve a purpose, but that's just a suggestion